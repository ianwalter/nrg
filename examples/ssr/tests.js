const { test } = require('@ianwalter/bff')
const app = require('./fixtures/base')
const nrg = require('..')

const result = { song: 'Wave' }
const addData = (ctx, next) => {
  ctx.state.body = result
  return next()
}
const end = ctx => (ctx.status = 200)

if (!process.env.GITHUB_ACTION) { // FIXME: why did I do this again?
  test('addToSsr', async t => {
    const assertion = ctx => t.expect(ctx.state.ssr.song).toBe(result.song)
    app.get('/', addData, nrg.addToSsr, assertion, end)
    await app.test('/').get()
  })

  test('addToSsr with namespace', async ({ expect }) => {
    const assertion = c => expect(c.state.ssr.next.ok.song).toBe(result.song)
    app.get('/namespace', addData, nrg.addToSsr('next.ok'), assertion, end)
    await app.test('/namespace').get()
  })
}

test('SSR', async t => {
  const app = require('../examples/ssr')

  // Verify that the page is returned successfully when requesting the root
  // path.
  let response = await app.test('/').get()
  t.expect(response.status).toBe(200)
  t.expect(response.text).toMatchSnapshot()

  // Verify that a 404 Not Found is returned when requesting a path that
  // contains "not-found"
  response = await app.test('/not-found').get()
  t.expect(response.status).toBe(404)
})
