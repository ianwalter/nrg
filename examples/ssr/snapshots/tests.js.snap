// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SSR 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\">
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
    <link rel=\\"icon\\" type=\\"image/png\\" href=\\"/static/favicon.png\\">
    
    <style type=\\"text/css\\"></style>
  </head>
  <body>
    <div id=\\"app\\"><h1>SSR Example</h1>

<p>The current URL is /</p></div>
  <script src=\\"ssr.js\\"></script></body>
</html>
"
`;
